app/models/miq_worker/container_common.rb

```diff
@@ -30,6 +31,21 @@ def zone_selector
       {"#{Vmdb::Appliance.PRODUCT_NAME.downcase}/zone-#{MiqServer.my_zone}" => "true"}
     end
 
+    def resource_constraints
+      mem_threshold = self.class.worker_settings[:memory_threshold]
+      cpu_threshold = self.class.worker_settings[:cpu_threshold_percent]
+
+      return {} if !Settings.server.worker_monitor.enforce_resource_constraints || (mem_threshold.nil? && cpu_threshold.nil?)
+
+      {:limits => {}}.tap do |h|
+        h[:limits][:memory] = "#{mem_threshold / 1.megabyte}Mi" if mem_threshold
```

Comment:
I can't remember but does this work if they specify 1.gigabyte for the memory_threshold?

Reply:
Yup:
```
irb(main):002:0> 1.gigabyte / 1.megabyte
=> 1024
```

But it doesn't really matter as it's all converted when accessed through `.worker_settings`:
```
irb(main):010:0> Settings.workers.worker_base.ui_worker.memory_threshold
=> "1.gigabytes"
irb(main):011:0> MiqUiWorker.worker_settings[:memory_threshold]
=> 1073741824
```